lib/dialects/abstract/query-generator.js

```diff
@@ -1552,15 +1271,21 @@ var QueryGenerator = {
       }
     }
 
+    if (self._dialect.supports.cteQueries && options.cte) {
```

Comment:
Just init query to an empty string at the top of the function, and overwrite it with the cteQuery if applicable - avoids having to check for cte twice